### PR TITLE
add workflows, main.go and docker-compose.yml

### LINKED ISSUE #1747: Add a standalone processor template
### Feature description

To help our users get started with writing standalone processors it would be good to have a standalone processor template, similar to https://github.com/ConduitIO/conduit-connector-template/.

It's also important this is documented somewhere in our documentation site: https://conduit.io/docs/processors/standalone/building

### PR DESCRIPTION
**Pull request by AdamHaffar**
Closes https://github.com/ConduitIO/conduit/issues/1747.

### CODE REVIEW

**Comment by lovromazgon** [comment]
Generally looks ok, I'm figuring out some minor details, specifically what it makes sense to showcase as the default processor, probably one that is simply passing records through.

File: README.md [hunk 1]
```diff
@@ -1,65 +1,71 @@
...
+## Quick start
+
+1. Click [_Use this template_]() and clone your new repository.
+2. Initialize the repository using [`setup.sh`](https://github.com/ConduitIO/conduit-processor-template/blob/main/setup.sh) and commit your changes.
+   ```sh
+   ./setup.sh github.com/myusername/conduit-processor-myprocessor
+   git add -A
+   git commit -m "initialize repository"
+   ```
+3. Set up [automatic Dependabot PR merges](#automatically-merging-dependabot-prs).a
```
Comment (raulb) [inline README.md]:
```suggestion
3. Set up [automatic Dependabot PR merges](#automatically-merging-dependabot-prs).
```

File: README.md [hunk 2]
```diff
@@ -1,65 +1,71 @@
...
+
+With that, you're all set up and ready to start working on your processor! As a next step, we recommend that you 
+check out the [Conduit Processor SDK](https://github.com/ConduitIO/conduit-processor-sdk).
+
+## What's included?
+
+* Skeleton code for the processor and its configuration.
+* Example unit tests.
+* A [Makefile](/Makefile) with commonly used targets.
+* A [GitHub workflow](/.github/workflows/build.yml) to build the code and run the tests.
```
Comment (raulb) [inline README.md]: We don't seem to have a build.yml

File: README.md [hunk 3]
```diff
@@ -1,65 +1,71 @@
...
+* A [GitHub workflow](/.github/workflows/lint.yml) to run a pre-configured set of linters.
+* A [GitHub workflow](/.github/workflows/release.yml) which automatically creates a release when a tag is pushed.
+* A [Dependabot setup](/.github/dependabot.yml) which checks your dependencies for available updates and 
+[merges minor version upgrades](/.github/workflows/dependabot-auto-merge-go.yml) automatically.
```
Comment (raulb) [inline README.md:35]: We don't have this one either.

**Review by raulb — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (lovromazgon) [reply README.md]: Good catch that was renamed to `test.yml`, we need to update it in the connector template as well.

https://github.com/ConduitIO/conduit-connector-template/pull/52

File: README.md (hunk 3 above)
Comment (lovromazgon) [reply README.md:35]: This file is actually added as part of this PR (see https://github.com/ConduitIO/conduit-processor-template/blob/workflows/.github/workflows/dependabot-auto-merge-go.yml).

Comment (raulb) [reply README.md:35]: I see. I only checked this branch, and main. My recommendation is update the README on that PR directly so we don't provide incomplete information until that PR is shipped.

**Review by raulb — APPROVED** [review]
Only one nit left.

File: README.md (hunk 3 above)
Comment (lovromazgon) [reply README.md:35]: Both are in this PR 🤔

Comment (raulb) [reply README.md:35]: Oh weird! I completely missed it. I actually thought they were different branches 🤦🏻‍♂️

**Pull request merged**